design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,174 @@
+* Define the task functions of the ClusterLink operator.
+* Define the CRD (Custom Resource Definition) for the ClusterLink operator.
+
+## Non-Goals
+
+The following aspects are explicitly excluded and are out of scope in the current design:
+
+* This document focuses only on the deployment of ClusterLink to the K8s cluster.
+The deployment to another environment as VMs is out of scope.
+* Security considerations related to the creation of certificates for the fabric or each site by the fabric administrator are also beyond the scope of this document
+
+## Proposal
+
+The ClusterLink deployment will be done in the following steps:
+
+1. Deploy the cluster certificates as k8s secrets to the cluster.
+   The certificate include the public Fabric CA certificates and the site certificate that generated by the fabric administrator.
+2. Create the ClusterLink operator.
+3. Deploy the ClusterLink CRD to the operator ClusterLink deployments that includes:
+
+   * ClusterLink controlplane deployment, including controlplane-pod, controlplane-service, and RBAC roles.  
+   * ClusterLink dataplane deployment, including dataplane pods, whether single or multiple, along with the dataplane-service..
+   * ClusterLink external access point using load-balancer/node-port/gateway-API service.
+  
+Overall, the ClusterLink deployment stages are:
+
+<img src="deployment.png" width="800" height="400" alt="ClusterLink Deployment Stages"/>
+
+The operator will be created in a dedicated namespace and, by default, will be ```cl-operator-ns```. It will have privileged permissions, enabling it to create ClusterLink components within the dedicated namespace.
+The ClusterLink components will be created in ```clusterlink-system-ns```, and the control-plane will have privileged permissions to execute operations on the ClusterLink CRDs, such as peer, export, import, policy, etc.
+Additionally, the operator will create and update components in response to the ```clusterlink.yaml``` create or update actions. Once created or updated, the operator will not actively monitor the state of each component, assuming that any changes will be made only by privileged users.
+Furthermore, the operator will delete all components and the namespace of ClusterLink when the ```clusterlink.yaml``` file is deleted.
+
+The deployment file for the operator and example of ```clusterlink.yaml``` will be included in every project release.
+
+The ClusterLink CLI will be utilized to automate the deployment process.\
+For instance, the following commands can assist the fabric administrator in creating certificates in the directory ```~/.config/clustrlink/<fabric_name>```:
+
+    clusterlink create fabric --name <fabric_name>
+    clusterlink create site --name <site_name> --fabric <fabric_name> 
+
+Automation for deploying the certificates as a secret to the cluster, deploying the ClusterLink operator, and the ```clusterlink.yaml``` can be achieved with
+
+    clusterlink install --site <site_name>
```

Comment:
How will you make the cluster kubeconfig accessible to the CLI?
What is the meaning of `site` here?

Reply:
So be explicit about that. Or at lease pause to get user acceptance when deploying to the last used kubeconfig. Another option is to explicitly require the kubeconfig and context as parameters.